webapp/packages/plugin-sql-editor-navigation-tab-resource/src/SqlEditorTabResourceService.ts

```diff
@@ -6,49 +6,122 @@
  * you may not use this file except in compliance with the License.
  */
 
+import { makeObservable, observable } from 'mobx';
+
 import { ITab, NavigationTabsService, NavTreeResource, NodeManagerUtils } from '@cloudbeaver/core-app';
 import { injectable } from '@cloudbeaver/core-di';
 import { NotificationService } from '@cloudbeaver/core-events';
 import { WindowEventsService } from '@cloudbeaver/core-root';
 import { ResourceKey, ResourceKeyUtils } from '@cloudbeaver/core-sdk';
+import { LocalStorageSaveService } from '@cloudbeaver/core-settings';
 import { throttleAsync } from '@cloudbeaver/core-utils';
 import { NavResourceNodeService } from '@cloudbeaver/plugin-resource-manager';
-import { ISqlEditorTabState, SqlEditorService } from '@cloudbeaver/plugin-sql-editor';
+import { IQueryChangeData, ISqlEditorTabState, SqlEditorService } from '@cloudbeaver/plugin-sql-editor';
 import { isSQLEditorTab, SqlEditorTabService } from '@cloudbeaver/plugin-sql-editor-navigation-tab';
 
 const SYNC_DELAY = 5 * 1000;
+const VALUE_SYNC_DELAY = 0.5 * 1000;
+const RESOURCE_TAB_STATE = 'sql_editor_resource_tab_state';
+
+interface IResourceTabData {
+  nodeId: string;
+  parents: string[];
+}
+
+type ResourceTabState = Map<string, IResourceTabData>;
 
 @injectable()
 export class SqlEditorTabResourceService {
+  state: ResourceTabState;
+
   constructor(
     private readonly navigationTabsService: NavigationTabsService,
     private readonly sqlEditorService: SqlEditorService,
     private readonly sqlEditorTabService: SqlEditorTabService,
     private readonly notificationService: NotificationService,
     private readonly navTreeResource: NavTreeResource,
     private readonly navResourceNodeService: NavResourceNodeService,
-    private readonly windowEventsService: WindowEventsService
+    private readonly windowEventsService: WindowEventsService,
+    private readonly localStorageSaveService: LocalStorageSaveService,
   ) {
+    this.state = new Map();
+
     this.onNodeDeleteHandler = this.onNodeDeleteHandler.bind(this);
     this.onTabSelectHandler = this.onTabSelectHandler.bind(this);
+    this.onTabCloseHandler = this.onTabCloseHandler.bind(this);
     this.onFocusChangeHandler = this.onFocusChangeHandler.bind(this);
+    this.onTabResourceValueChangeHandler = this.onTabResourceValueChangeHandler.bind(this);
 
     this.start = this.start.bind(this);
     this.stop = this.stop.bind(this);
 
     this.syncTab = throttleAsync(this.syncTab, SYNC_DELAY);
+    this.syncTabResourceValue = throttleAsync(this.syncTabResourceValue, VALUE_SYNC_DELAY);
```

Comment:
`syncTab` -> `updateTabQuery`
`syncTabResourceValue` -> `updateResource`

Reply:
also for `updateResource` we need a throttle that reset timeout each time it is called, so script save should be executed only if in the last N ms delay will be no executions